File: C/0441.c
// attempt1 create rows
int arrangeCoins(int n) {
    int rows = 0, i = 1;

    while (n >= i) {
        n -= i;
        i++;
        rows++;
    }

    return rows;
}

// attempt2 single calculation
#include <math.h>

int arrangeCoins(int n) {
    return (int)(-1 + sqrt(1 + 8.0 * n)) / 2;
}

File: Go/0441.go
// attempt1 create rows
func arrangeCoins(n int) int {
    rows := 0
    i := 1

    for n >= i {
        n -= i
        i++
        rows++
    }

    return rows
}

// attempt2 single calculation
import "math"

func arrangeCoins(n int) int {
    return (int)(-1 + math.Sqrt((float64)(1 + 8 * n))) / 2
}
